pkg/tbtc/dkg.go

```diff
@@ -296,14 +312,14 @@ func (de *dkgExecutor) generateSigningGroup(
 
 					// Set up the attempt timeout signal.
```

Comment:
Given that we moved all attempt-related timeouts to `dkg_loop.go`, would it be cleaner if we passed the timeout block as a parameter in `dkgAttemtParams`? We do this for signing:

https://github.com/keep-network/keep-core/blob/71e0ac8fb1cd6a63994d897216a2d85764fe85a7/pkg/tbtc/signing_loop.go#L282

Also, not something we should solve in this PR, but I think it's worth opening an issue if you agree: instead of passing timeout block we should pass a context. The context should be cancelled inside `dkg_loop.go` / `signing_loop.go`.

Right now, the retry loop assumes the attempt should start at the given block:
https://github.com/keep-network/keep-core/blob/71e0ac8fb1cd6a63994d897216a2d85764fe85a7/pkg/tbtc/signing_loop.go#L182-L185

but it can't guarantee the previous attempt ended before. It has to trust `signingAttemptFn`. The same situation is for DKG.

Reply:
See d659537d. Regarding:

> instead of passing timeout block we should pass a context

I don't think it's worth our time now. I think so because:
- the current shape is good enough (leave well alone!) and this mechanism is quite complex. We may not be aware of all implications of this change right now
- the current approach has clear and balanced boundaries: the loop just determines the start and timeout blocks and does not care what `signingAttemptFn` does. After all, `signingAttemptFn` triggers the actual action and has the full knowledge to clean up it properly. Using a context prepared within the loop is less flexible and will push much more responsibilities on the loop side.
- the current approach is easier to test as we can easily check whether the blocks proposed by the loop makes sense for the given attempt